libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -908,6 +914,14 @@ YAML::Node YamlConfiguration::getOptionalField(const YAML::Node& yamlNode, const
   return result;
 }
 
+void YamlConfiguration::addNewId(const std::string& uuid) {
+  if (uuids_.find(uuid) != uuids_.end()) {
+    throw Exception(ExceptionType::GENERAL_EXCEPTION, "UUID " + uuid + " is duplicated in the flow configuration");
+  }
+
+  uuids_.insert(uuid);
```

Comment:
minor, but `insert()` returns whether the insertion was successful, so we could avoid doing two lookups:
```suggestion
  const auto [_, success] = uuids_.insert(uuid);
  if (!success) {
    throw Exception(ExceptionType::GENERAL_EXCEPTION, "UUID " + uuid + " is duplicated in the flow configuration");
  }
```

Reply:
Good idea, updated in 33c6c28c702f72e67add5c9dfd318020019a1788